src/psyclone/domain/lfric/transformations/lfric_extract_trans.py

```diff
@@ -147,20 +147,20 @@ def apply(self, nodes, options=None):
         region_name = self.get_unique_region_name(nodes, my_options)
         my_options["region_name"] = region_name
         my_options["prefix"] = my_options.get("prefix", "extract")
-        input_list, output_list = dep.get_in_out_parameters(nodes)
+        # TODO
+        read_write_info = dep.get_in_out_parameters(nodes)
+        my_options["read_write_info"] = read_write_info
```

Comment:
Yes, and no - the issue is that the generic PSyData transformation class creates the specific PSyData node (e.g. ExtractNode). So we need to find a way to pass the list of variables from the transformation to the PSyIR node via the apply in PSyData:
```
        # We have to create an instance of the node that will be inserted in
        # order to find out what module name it will use.
        pdata_node = self._node_class(options=my_options)
```
The option dictionary seems to be the right way of doing this.

Reply:
I've moved the assignment to the option in lfric_extract_trans and added a comment to make it clearer why this is done. I hope that's enough.

Reply:
Yes, I think that helps, thanks :-)